### PR TITLE
Refactor SQL filtering

### PR DESCRIPTION
**Pull request by rgwozdz**
This PR:

* put methods into their own files and put under direct unit test
* fixes a bug in the "envelope-intersects" operation.  The old code was checking for an intersection of the feature envelope against the query geometry.  However, research indicates that the Esri envelope-intersects operation should check that an "Envelope of Query Geometry Intersects Envelope of Target Geometry" (see [here](http://resources.esri.com/help/9.3/ArcGISDesktop/ArcObjects/esriGeoDatabase/esriSpatialRelEnum.htm)).
* adds support for the `!=` operator in the hashed OBJECTID comparison.

### CODE REVIEW

File: lib/filter-and-transform/filters/contains.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const { contains } = require('@terraformer/spatial')
+module.exports = function (featureGeometry = {}, filterGeometry = {}) {
+  if (!featureGeometry) return false
```
Comment (haoliangyu) [inline lib/filter-and-transform/filters/contains.js:3]: Use `_.isEmpty` because the falsy value has defaulted to `{}`.

File: lib/filter-and-transform/filters/intersects.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const { intersects, contains } = require('@terraformer/spatial')
+const convertFromEsri = require('../../geometry/transfrom-esri-geometry-to-geojson-geometry')
+
+module.exports = function (featureGeometry = {}, filterGeometry = {}) {
+  if (!featureGeometry) return false
```
Comment (haoliangyu) [inline lib/filter-and-transform/filters/intersects.js:5]: Use `_.isEmpty`

File: lib/filter-and-transform/filters/within.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const { within } = require('@terraformer/spatial')
+
+module.exports = function (featureGeometry, filterGeometry = {}) {
+  if (!featureGeometry) return false
```
Comment (haoliangyu) [inline lib/filter-and-transform/filters/within.js:4]: Use `_.isEmpty`

File: test/unit/filter-and-transform/filters/constains.spec.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+const contains = require('../../../../lib/filter-and-transform/filters/contains')
+
+test('contains: empty input', t => {
+  const result = contains()
+  t.equals(result, false)
+  t.end()
+})
+
+test('contains: empty object input', t => {
+  const result = contains({}, {})
```
Comment (haoliangyu) [inline test/unit/filter-and-transform/filters/constains.spec.js:11]: Probably adding tests for `contains(null, {})` and `contains({}, null)` since you are validating the two parameters separately.

**Review by haoliangyu — CHANGES_REQUESTED** [review]

**Comment by rgwozdz** [comment]
@haoliangyu - requested changes were made in e76946e.

**Review by haoliangyu — APPROVED** [review]
